app/controllers/admin/settings/project_custom_fields_controller.rb

```diff
@@ -75,15 +75,15 @@ def project_mappings
 
     def link
       create_service = ProjectCustomFieldProjectMappings::BulkCreateService
-                         .new(user: current_user, project: @project, project_custom_field: @custom_field,
+                         .new(user: current_user, projects: @projects, project_custom_field: @custom_field,
```

Comment:
> What do you think about passing project_ids to the service and moving all logic about finding and initializing relevant projects there? In current implementation this logic is distributed between controller and service. I suggest to avoid that.

I reckon one disadvantage to that approach is that we can't fail early- and return a 404 if some invalid data is passed. The controller is nicely updating the flash component via turbo in this case- which makes a case for keeping it in the controller. WDYT?

Reply:
> And do we need to react with `Not found` error in case there are none projects with given ids? Then we could simplify logic a bit more.

Not necessarily, just handling a corner case. Given it's just a one liner guard clause, I reckoned that was acceptable?

Reply:
Yes, I see. My point if it is really needed to handle this corner case. If it is very unlikely to happen then maybe just respond with 204 no content and do nothing. Let's have a chat.